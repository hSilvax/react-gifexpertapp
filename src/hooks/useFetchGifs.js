import {useState, useEffect} from 'react'
import { getGifts } from '../helpers/getGifs';

export const useFetchGifs = (category)=> {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(()=> {
        
        getGifts(category)
            .then( imgs => {
            

                setState({
                    data: imgs,
                    loading: false
                });                             

            })


    }, [category]); // [] solo se ejecuta una vez */
   

    return state;

}