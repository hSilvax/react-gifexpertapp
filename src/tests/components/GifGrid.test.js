import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas de <GifGrid/>", () => {
    
    const category = "Macross";    

    test("Debe mostrarse bien el control", () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test("Debe de mostrar item cuando se carga imagenes useFetchGifs", () => {
        
        const gifs = [{
            id: "ABC",
            url: "https//localhost/xxxx/xxxx.jpg",
            title: "XXXXXXXX"
        },
        {
            id: "ABC",
            url: "https//localhost/xxxx/xxxx.jpg",
            title: "XXXXXXXX"
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);

    });
    
    

})
