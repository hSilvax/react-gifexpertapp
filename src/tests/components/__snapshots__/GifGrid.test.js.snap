// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid/> Debe de mostrar item cuando se carga imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Macross
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="XXXXXXXX"
      url="https//localhost/xxxx/xxxx.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="XXXXXXXX"
      url="https//localhost/xxxx/xxxx.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de <GifGrid/> Debe mostrarse bien el control 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Macross
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
